deploy/sre-pruning/105-pruning.rbac.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: sre-pruner-sa
+  namespace: openshift-sre-pruning
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: sre-pruner-buildsdeploys-cr
+rules:
+# deployment pruning
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  - replicationcontrollers
+  verbs:
+  - list
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - replicationcontrollers
+  verbs:
+  - delete
+- apiGroups:
+  - apps.openshift.io
+  resources:
+  - deploymentconfigs
+  verbs:
+  - get
+  - list
+# builds pruning
+- apiGroups:
+  - build.openshift.io
+  resources:
+  - buildconfigs
+  - builds
+  verbs:
+  - list
+  - get
+- apiGroups:
+  - build.openshift.io
+  resources:
+  - builds
+  verbs:
+  - delete
```

Comment:
Does `delete` work without also including `list` and `get`?

Reply:
It might work without `get`, but it does need `list`. Here's a (somewhat redacted) sample of the prune transaction to show the API calls:

```lseelye@molydbenum sre-pruning $ oc adm prune builds --keep-complete=0 --keep-failed=0 --keep-younger-than=2m -v7 --confirm
I0405 09:31:00.867020   19262 loader.go:359] Config loaded from file ~/.kube/cluster
I0405 09:31:00.870674   19262 round_trippers.go:383] GET https://api.mycluster.example.com:6443/apis/build.openshift.io/v1/buildconfigs
I0405 09:31:00.870718   19262 round_trippers.go:390] Request Headers:
I0405 09:31:00.870734   19262 round_trippers.go:393]     Accept: application/json, */*
I0405 09:31:00.870742   19262 round_trippers.go:393]     User-Agent: oc/v1.11.0+d4cacc0 (darwin/amd64) kubernetes/d4cacc0
I0405 09:31:00.870748   19262 round_trippers.go:393]     Authorization: Bearer QV6JEXXrj2v-lCyxBJtuqAZePE_2OVmVSl1dPdOlV7I
I0405 09:31:01.397836   19262 round_trippers.go:408] Response Status: 200 OK in 527 milliseconds
I0405 09:31:01.401416   19262 round_trippers.go:383] GET https://api.mycluster.example.com:6443/apis/build.openshift.io/v1/builds
I0405 09:31:01.401448   19262 round_trippers.go:390] Request Headers:
I0405 09:31:01.401458   19262 round_trippers.go:393]     User-Agent: oc/v1.11.0+d4cacc0 (darwin/amd64) kubernetes/d4cacc0
I0405 09:31:01.401464   19262 round_trippers.go:393]     Authorization: Bearer QV6JEXXrj2v-lCyxBJtuqAZePE_2OVmVSl1dPdOlV7I
I0405 09:31:01.401471   19262 round_trippers.go:393]     Accept: application/json, */*
I0405 09:31:01.515934   19262 round_trippers.go:408] Response Status: 200 OK in 114 milliseconds
I0405 09:31:01.521470   19262 prune.go:49] Creating build pruner with keepYoungerThan=2m0s, orphans=false, keepComplete=0, keepFailed=0
I0405 09:31:01.521942   19262 round_trippers.go:383] DELETE https://api.mycluster.example.com:6443/apis/build.openshift.io/v1/namespaces/default/builds/ruby-sample-build-1
I0405 09:31:01.521964   19262 round_trippers.go:390] Request Headers:
I0405 09:31:01.521975   19262 round_trippers.go:393]     Accept: application/json, */*
I0405 09:31:01.521985   19262 round_trippers.go:393]     Content-Type: application/json
I0405 09:31:01.521996   19262 round_trippers.go:393]     Authorization: Bearer QV6JEXXrj2v-lCyxBJtuqAZePE_2OVmVSl1dPdOlV7I
I0405 09:31:01.522006   19262 round_trippers.go:393]     User-Agent: oc/v1.11.0+d4cacc0 (darwin/amd64) kubernetes/d4cacc0
I0405 09:31:01.639509   19262 round_trippers.go:408] Response Status: 200 OK in 117 milliseconds
NAMESPACE   NAME
default     sample-build-1```